src/Handler/LogDnaHandler.php

```diff
@@ -68,6 +70,28 @@ protected function getDefaultFormatter(): FormatterInterface
 
     public function write(array $record)
     {
+        $body = $record['formatted'];
+
+        if (mb_strlen($body) > static::LOGDNA_BYTE_LIMIT) {
+            $decodedBody = json_decode($body, true);
+
+            $body = json_encode([
+                'lines' => [
+                    [
+                        'timestamp' => $decodedBody['lines'][0]['timestamp'] ?? '',
+                        'line'      => $decodedBody['lines'][0]['line'] ?? '',
+                        'app'       => $decodedBody['lines'][0]['app'] ?? '',
+                        'level'     => $decodedBody['lines'][0]['level'] ?? '',
+                        'meta'      => [
+                            'longException' => mb_substr($body, 0, static::LOGDNA_BYTE_LIMIT),
+                        ],
+                    ],
+                ],
+            ]);
+        }
+
+        $this->lastBody = $body;
```

Comment:
Is this used anywhere outside of the tests? If not, can we remove it?

Reply:
I added it specifically for tests, not sure how i'd test without i

Reply:
Ok yeah I see what you mean. I have no better solution then 😄